### PR TITLE
Ensure tckTests run

### PR DESCRIPTION
**Pull request by rozza**
JAVA-4266

### CODE REVIEW

File: driver-reactive-streams/build.gradle [hunk 1]
```diff
@@ -37,6 +37,7 @@ dependencies {
 
 sourceSets {
     test.groovy.srcDirs += ['src/examples']
+    test.java.srcDirs += ['src/test/tck']
```
Comment (rozza) [inline driver-reactive-streams/build.gradle:40]: This was the cause

File: driver-reactive-streams/build.gradle [hunk 2]
```diff
@@ -51,10 +52,10 @@ task tckTest(type: Test) {
     classpath = sourceSets.test.runtimeClasspath
     testClassesDirs = sourceSets.test.output.classesDirs
 
-    binResultsDir = file("$buildDir/$name-results/binary")
+    binaryResultsDirectory.set(file("$buildDir/$name-results/binary"))
```
Comment (rozza) [inline driver-reactive-streams/build.gradle:55]: deprecation fixes

File: driver-reactive-streams/src/test/tck/com/mongodb/reactivestreams/client/ListDatabasesPublisherVerification.java [hunk 3]
```diff
@@ -45,8 +46,7 @@ public ListDatabasesPublisherVerification() {
         for (long i = 0; i < elements; i++) {
             run(client.getDatabase(getDefaultDatabaseName() + i).createCollection("test" + i));
         }
-
-        return client.listDatabases().filter(Filters.nin("name", "admin", "config", "local"));
+        return client.listDatabases().filter(Filters.regex("name", format("^%s.*", getDefaultDatabaseName())));
```
Comment (rozza) [inline driver-reactive-streams/src/test/tck/com/mongodb/reactivestreams/client/ListDatabasesPublisherVerification.java:49]: Slight tweak - ensures the databases list only includes the created databases.

File: driver-reactive-streams/src/test/tck/com/mongodb/reactivestreams/client/PublishersVerification.java [hunk 4]
```diff
@@ -36,10 +35,7 @@ public PublishersVerification() {
     @Override
     public Publisher<Integer> createPublisher(final long elements) {
         assert (elements <= maxElementsFromPublisher());
-
-        return Publishers.publishAndFlatten(callback ->
```
Comment (rozza) [inline driver-reactive-streams/src/test/tck/com/mongodb/reactivestreams/client/PublishersVerification.java:40]: These old publishers no longer exist - so use the Flux equivalents

**Review by jyemin — APPROVED** [review]
I don't fully understand how this works, but it does seem to work!

LGTM

**Pull request merged**